pkgs/development/python-modules/pyqt/6.x.nix

```diff
@@ -0,0 +1,134 @@
+{ lib
+, buildPythonPackage
+, isPy27
+, fetchPypi
+, pkg-config
+, dbus
+, lndir
+, dbus-python
+, sip
+, pyqt6-sip
+, pyqt-builder
+, qt6Packages
+, withConnectivity ? false
+, withMultimedia ? false
+, withWebKit ? false
+, withWebSockets ? false
+, withLocation ? false
+}:
+
+buildPythonPackage rec {
+  pname = "PyQt6";
+  version = "6.3.1";
+  format = "pyproject";
+
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-jMbiHbr3BH0fyJfjlszZcQoS8u+XZWPa1l8GAX0sl1c=";
+  };
+
+  patches = [
+    # Fix some wrong assumptions by ./project.py
+    # TODO: figure out how to send this upstream
+    # ./pyqt5-fix-dbus-mainloop-support.patch
+    # confirm license when installing via pyqt6_sip
+    ./pyqt5-confirm-license.patch
+  ];
+
+  # be more verbose
+  postPatch = ''
+    cat >> pyproject.toml <<EOF
+    [tool.sip.project]
+    verbose = true
+    EOF
+  '';
+
+  postUnpack = ''
+  echo QMAKEPATH=$QMAKEPATH
+
+  # paralellize compilation of , which happens within ./configure
+  export MAKEFLAGS+=" -j$NIX_BUILD_CORES"
+  '';
```

Comment:
Should not be required with the qmake setupHook.

Reply:
Wasn't working for the make calls inside some python code in here, afaict. This made it massively faster to test PyQt6 changes.

Reply:
I'll add a note that this is hacky/shouldn't be needed

Reply:
Documented this

Reply:
#186874 - fix for pip-build-hook.sh that makes this hacky bit unnecessary.

Rebuilds 5000+ :(